src/Evolution/Systems/Cce/Tags.hpp

```diff
@@ -0,0 +1,244 @@
+
+/// A prefix tag representing a quantity that will appear on the right-hand side
+/// of an explicitly regular differential equation
+template <typename Tag>
+struct Integrand : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "Integrand(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing the boundary data for a quantity on the extraction
+/// surface.
+template <typename Tag>
+struct BoundaryValue : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "BoundaryValue(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing the coefficient of a pole part of the right-hand
+/// side of a singular differential equation
+template <typename Tag>
+struct PoleOfIntegrand : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "PoleOfIntegrand(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing the regular part of the right-hand side of a
+/// regular differential equation
+template <typename Tag>
+struct RegularIntegrand : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "RegularIntegrand(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing a linear operator that acts on `Tag`. To determine
+/// the spin weight, It is assumed that the linear operator plays the role of
+/// \f$L\f$ in an equation of the form,
+/// \f$ (y - 1) \partial_y H + L H + L^\prime \bar{H} = A + (1 - y) B \f$
+template <typename Tag>
+struct LinearFactor : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, 0>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "LinearFactor(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing a linear operator that acts on `Tag`. To determine
+/// the spin weight, it is assumed that the linear operator plays the role of
+/// \f$L^\prime\f$ in an equation of the form,
+/// \f$ (y - 1) \partial_y H + L H + L^\prime \bar{H} = A + (1 - y) B \f$
+template <typename Tag>
+struct LinearFactorForConjugate : db::PrefixTag, db::SimpleTag {
+  using type =
+      Scalar<SpinWeighted<ComplexDataVector, 2 * Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "LinearFactorForConjugate(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing the product of two other tags, needed to
+/// represent some of the cached quantities for CCE. Note that the spins are
+/// added.
+template <typename LhsTag, typename RhsTag>
+struct Multiplies : db::PrefixTag, db::SimpleTag {
```

Comment:
I suggested to move this to SpinWeighted tags, since it’s not specific to CCE so shouldn’t be in this namespace. Did you disagree with the suggestion?

Reply:
Sorry, I incorrectly thought I had responded to this suggestion. While it is true that it isn't strictly specific to CCE, the more complicated assembly of tags that this is used for are less likely to be used generally. I'd prefer to keep it limited to the CCE scope until we find a need elsewhere.

Reply:
Alright, that's reasonable. I'll move it to `DataStructures/Tags.hpp`.